### ISSUE TITLE
`no-undefined-types` error when casting a generic template

### ISSUE THREAD

**Issue by jespertheend** [gajus/eslint-plugin-jsdoc#827]
## Expected behavior

No error when casting generic types.

## Actual behavior

`The type 'T' is undefined. (jsdoc/no-undefined-types)`

## ESLint Config

```js
module.exports = {
	root: true,
	parser: "@babel/eslint-parser",
	parserOptions: {
		requireConfigFile: false,
	},
	env: {
		browser: true,
		es2021: true,
	},
	extends: "eslint:recommended",
	plugins: [
		"jsdoc",
	],
	settings: {
		jsdoc: {
			mode: "typescript",
		},
	},
	rules: {
		"jsdoc/no-undefined-types": ["error"],
	},
};

```

## ESLint sample

```js
/**
 * @template T
 */
export class Foo {
	// cast to T
	getType() {
		const x = "hello";
		const y = /** @type {T} */ (x);
		return y;
	}
}
```

## Environment

- Node version: `v16.13.0`
- ESLint version: `v8.2.0`
- eslint-plugin-jsdoc version: `37.4.0`

**Comment by jespertheend** [comment]
Hmm I think this might be a duplicate of https://github.com/gajus/eslint-plugin-jsdoc/issues/559, not sure
